frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -10,74 +9,94 @@ import {
-import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
-import { getName } from '@console/shared';
+import { getName, getRequestedPVCSize } from '@console/shared';
 import { OCSServiceModel } from '../../../models';
 import { OSD_CAPACITY_SIZES } from '../../../utils/osd-size-dropdown';
-import { CEPH_STORAGE_NAMESPACE, NO_PROVISIONER } from '../../../constants';
-import { labelTooltip, storageClassTooltip } from '../../../constants/ocs-install';
-import { CAPACITY_USAGE_QUERIES, StorageDashboardQuery } from '../../../constants/queries';
+import { NO_PROVISIONER } from '../../../constants';
+import {
+  labelTooltip,
+  storageClassTooltip,
+  defaultRequestSize,
+} from '../../../constants/ocs-install';
 import { OCSStorageClassDropdown } from '../storage-class-dropdown';
 import { PVsAvailableCapacity } from '../../ocs-install/pvs-available-capacity';
 import './_add-capacity-modal.scss';
+import { createDeviceSet, DeviceSet } from '../../ocs-install/ocs-request-data';
+import { cephCapacityResource } from '../../../constants/resources';
 
 const getProvisionedCapacity = (value: number) => (value % 1 ? (value * 3).toFixed(2) : value * 3);
 
+const getCurrentDeviceSet = (deviceSets: DeviceSet[], selectedSCName: string): number =>
+  deviceSets.findIndex((ds) => ds.dataPVCTemplate.spec.storageClassName === selectedSCName);
+
 export const AddCapacityModal = (props: AddCapacityModalProps) => {
   const { ocsConfig, close, cancel } = props;
+  const deviceSets: DeviceSet[] = ocsConfig?.spec?.storageDeviceSets || [];
 
-  const [response, loadError, loading] = usePrometheusPoll({
-    endpoint: PrometheusEndpoint.QUERY,
-    namespace: CEPH_STORAGE_NAMESPACE,
-    query: CAPACITY_USAGE_QUERIES[StorageDashboardQuery.CEPH_CAPACITY_USED],
-  });
+  const [response, loadError, loading] = usePrometheusPoll(cephCapacityResource);
   const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
+  /* TBD(Afreen): Show installation storage class as preselected 
+                  Change state metadata
+  */
   const [inProgress, setProgress] = React.useState(false);
   const [errorMessage, setError] = React.useState('');
 
-  const osdSizeWithUnit = _.get(
-    ocsConfig,
-    'spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage',
-  );
-  const presentCount = _.get(ocsConfig, 'spec.storageDeviceSets[0].count');
-  const capacity = _.get(response, 'data.result[0].value[1]');
-
-  const osdSizeWithoutUnit = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
+  const cephCapacity: string = response?.data?.result?.[0]?.value[1];
+  const osdSizeWithUnit = getRequestedPVCSize(deviceSets[0].dataPVCTemplate);
+  const osdSizeWithoutUnit: number = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
   const provisionedCapacity = getProvisionedCapacity(osdSizeWithoutUnit);
+  const isNoProvionerSC: boolean = storageClass?.provisioner === NO_PROVISIONER;
```

Comment:
```suggestion
  const isNoProvionerSC = storageClass?.provisioner === NO_PROVISIONER;
```
Implicit from the boolean comparison

Reply:
I understand, but I prefer to read that as well :)